### ISSUE TITLE
Uncaught Error: Can't resolve all parameters for e: (?)

### ISSUE THREAD

**Issue by LuisFlip** [angular-fullstack/generator-angular-fullstack#2770]
- [x] I understand that GitHub issues are not for tech support, but for questions specific to this generator, bug reports, and feature requests.

Item | Version
----- | -----
generator-angular-fullstack | 5.0.0 RC4
Node | 8.12.0
npm | 6.4.1
Operating System | Windows 10

Item | Answer
----- | -----
Transpiler |  TypeScript
Markup | HTML 
CSS | SCSS 
Client Tests |  Mocha
DB | MongoDB 
Auth | Y 

After create project using the generator, i built it in production environment and when i load the app  the follow error appears.

![image](https://user-images.githubusercontent.com/39152997/47425229-a0016880-d781-11e8-846c-70f7362f11ad.png)

**Comment by halit-rs** [comment]
I resolve this issue with uncomment  `import 'core-js/es7/reflect';` in pollyfills.ts.

**Comment by LuisFlip** [comment]
i run:
`bulp build`
`cd dist`
`node server/index.js`
next open `localhost:8080`
and get the error

**Comment by Awk34** [comment]
I was able to reproduce and fix this locally. Apparently it's an issue with webpack bundling.

Manually add the following to the top of your `polyfills.ts` file:
```ts
import 'reflect-metadata';
```

**Comment by Awk34** [comment]
See https://github.com/NativeScript/nativescript-angular/issues/1540 about the same issue

**Comment by LuisFlip** [comment]
i try 
` import 'reflect-metadata'; `
`// import 'core-js/es6';`
`// import 'core-js/es7/reflect';`
`import 'zone.js/dist/zone';`
`//import 'core-js/es7/reflect'; `
and get the error. i don't have the `nativescript-angular` in my project.

how i build project with AOT?

**Comment by Awk34** [comment]
It has nothing to do with that project; they just had a similar issue.

I was able to reproduce this, but adding the `reflect-metadata` import fixed the issue for me. Can you provide some more details about your case? Perhaps replace these lines: https://github.com/angular-fullstack/generator-angular-fullstack/blob/master/templates/app/webpack.make.js#L29-L31 with this:
```ts
config.mode = 'development';
```
and try building again, so we can see the un-minified error.

**Comment by LuisFlip** [comment]
Replacing that lines fixed the issue.

Before the change, the dist/client files created by build have the following sizes:  
![image](https://user-images.githubusercontent.com/39152997/47710066-b5670e80-dc29-11e8-94ce-7ae70c9d0d62.png)
After that replace, the build lines
![image](https://user-images.githubusercontent.com/39152997/47710071-b8fa9580-dc29-11e8-955b-4a15cd608826.png)

why we a have three warnings because of the size (1.3~mb) and after the changes we got higher values but not showing any warning?

is this in AOT or JIT?

best regards and thanks

**Comment by Awk34** [comment]
That change isn't meant to fix the issue, just turn off minification so that we can debug the error better, but if the error goes away when you set Webpack to development mode, then it has to have something to do with minification for builds.

**Comment by koraysels** [comment]
yeah the error ahs something to do with the reflect polyfill.. I have the same issue.

**Comment by koraysels** [comment]
I think this is a problem with UglifyJs. If you set the webpack mode to 'production' it enabels UglifyJsPlugin and then it fails.. Had anybody found a solution for this?

**Comment by Sina7312** [comment]
I did this at top of my `app.ts` file, and it solve the problem:
(commented out polyfills import, and add imports for reflect and zone.js)

```
import 'reflect-metadata';
import 'zone.js/dist/zone';
// import './polyfills';
```

**Comment by LuisFlip** [comment]
@Awk34  I solve the problem adding reflect-metadata on top of app.ts
`app.ts`:
```
import 'reflect-metadata';
import 'zone.js/dist/zone';
//import './polyfills';
```

**Comment by LuisFlip** [comment]
@koraysels I replace the uglify-js to terser.
https://github.com/mishoo/UglifyJS2/issues/659#issuecomment-447820525
if you want continues using the uglify-js you need install UglifyJS2#harmony-v2.8.22. see: https://github.com/webpack-contrib/uglifyjs-webpack-plugin/tree/version-0.4#install

**Comment by tschaffter** [comment]
Moving the imports from `polyfills.ts` to `app.ts` solves the issue as suggested here by @LuisFlip.

The Uglifier is not the issue (the issue is still there when commenting it in `webpack.make.js`).

## Solution
In `webpack.make.js`, declare `polyfills` before `app` in `config.entry`.

Before:
```
    if (!TEST) {
        config.entry = {
            polyfills: './client/app/polyfills.ts',
            vendor: [
                'lodash'
            ],
            app: './client/app/app.ts'
        };
    }
```

Now:
```
    if (!TEST) {
        config.entry = {
            polyfills: './client/app/polyfills.ts',
            vendor: [
                'lodash'
            ],
            app: './client/app/app.ts'
        };
    }
```

Here is what I have included at the top of `./poyfills.ts`:

```
// Enable certain polyfills depending on which browsers you need to support
import 'core-js/es6';
import 'core-js/es7/reflect';
import 'zone.js/dist/zone';
```

**Comment by atxiaoxian** [comment]
@tschaffter it solved the problem for me
